pkg/services/govmomi/service.go

```diff
@@ -530,17 +530,10 @@ func (vms *VMService) getMetadata(ctx *virtualMachineContext) (string, error) {
 
 	var metadataBase64 string
 	for _, ec := range obj.Config.ExtraConfig {
-		if optVal := ec.GetOptionValue(); optVal != nil {
-			// TODO(akutz) Using a switch instead of if in case we ever
-			//             want to check the metadata encoding as well.
-			//             Since the image stamped images always use
-			//             base64, it should be okay to not check.
-			//nolint:gocritic
-			switch optVal.Key {
-			case guestInfoKeyMetadata:
-				if v, ok := optVal.Value.(string); ok {
-					metadataBase64 = v
-				}
+		if optVal := ec.GetOptionValue(); optVal != nil && optVal.Key == guestInfoKeyMetadata {
+			if v, ok := optVal.Value.(string); ok {
+				metadataBase64 = v
+				break
```

Comment:
```suggestion
```

We should not break here.

If `guestInfoKeyMetadata` is multiple times in `ExtraConfig`, the last entry wins. If we break here the first wins.

So this would be a breaking change.

(I did not dig into if it is possible that we have multiple entries).

Reply:
You're right. I think it's best to preserve it as is, even if it seems unlikely there's a reasonable config that includes this multiple times.

Reply:
Make sense. I will remove the `break`.